### ISSUE TITLE
Brownie install failed

### ISSUE THREAD

**Issue by JMiddey** [smartcontractkit/full-blockchain-solidity-course-py#527]
pipx install eth-brownie
Fatal error from pip prevented installation. Full pip output in file:
    /Users/joe/.local/pipx/logs/cmd_2021-12-09_17.25.53_pip_errors.log

pip failed to build package:
    cytoolz

Some possibly relevant errors from pip install:
    cytoolz/functoolz.c:23087:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    cytoolz/functoolz.c:23092:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    cytoolz/functoolz.c:23176:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    error: command '/usr/bin/clang' failed with exit code 1

Error installing eth-brownie.

**Comment by cromewar** [comment]
Hello @JMiddey 

if you are on windows you must install visual studio community and the python development tools on it.
if you are on MAC you must install Xcode and cython.

Best regards, Cromewar.

**Comment by JMiddey** [comment]
Thank you got that bit sorted now :)

**Comment by Jaqck** [comment]
I have installed the visual studio community and the python development tools and still the same error. I don't get this error if installing with pip instead of pipx but using pip leads to a PATH error once brownie is installed. Any help?

**Comment by cromewar** [comment]
Could you share here the path error please?

**Comment by Jaqck** [comment]
Yes no problem. As I stated above I've installed using pip and this is the path error I receive: 

eth-brownie : The term 'eth-brownie' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ eth-brownie --version
+ ~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (eth-brownie:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

This is after running the eth-brownies --version command after installation.

**Comment by cromewar** [comment]
jum strange, did you choosed the correct python interpreter on vscode? the one you installed eth-brownie using pip

**Comment by Jaqck** [comment]
Ii believe so. Is there anyway I can check this without uninstalling and reinstalling?

**Comment by Jaqck** [comment]
I have found a few videos on this topic so I continued to attempt to troubleshoot. I have two python versions installed. When using the select interpreter tool I have attempted to use both installed and then run the eth-brownie --version command and still experience the same issue.

**Comment by cromewar** [comment]
Just as a try, could you please try with a conda env?

**Comment by Jaqck** [comment]
Some progress has been made! I completely uninstalled python 3.10 and installed python 3.9.8 instead. From here I am attempting to reinstall all of the programmes I previously had on python 3.10. However, pipx is currently trying to search for python in the old python3.10 directory that no longer exists. I hope after fixing this it may be smooth sailing

**Comment by Jaqck** [comment]
PS C:\Users\Jmclo\Documents\Demo\Python_Test> pipx install eth-brownie 
No Python at 'C:\Users\Jmclo\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\python.exe'
Fatal error from pip prevented installation. Full pip output in file:
    C:\Users\Jmclo\.local\pipx\logs\cmd_2022-01-06_18.15.09_pip_errors.log

Error installing eth-brownie.

**Comment by cromewar** [comment]
Try using regular pip instead pipx, as you will have more problems with vscode later if you install it using pipx.

**Comment by Jaqck** [comment]
I have installed using pip and pipx now and both return the same original error. 

eth-brownie : The term 'eth-brownie' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1

eth-brownie --version
  + CategoryInfo          : ObjectNotFound: (eth-brownie:String) [], CommandNotFoundException
  + FullyQualifiedErrorId : CommandNotFoundException

**Comment by cromewar** [comment]
@Jaqck it must be a problem with your windows console, here is some tips:

1. use powershell instead of cmd (windows terminal app might help)
2. use a miniconda environment.
3. your best choise is to use WSL (windows subsytem for linux) it will help you a lot.

**Comment by Jaqck** [comment]
I have managed to fix this issue thanks for the help :)

**Comment by cromewar** [comment]
Could you share the solution for the community?

**Comment by Jaqck** [comment]
Yes I believe the issue to have been some sort of error with the brownie installation. I uninstalled brownie, then set my environment variables and reinstalled. after installing without running any brownie commands I set a new environment variable into the brownie folder rather than just to the folder that had the brownie folder inside of it. I then ran brownie --version, brownie init etc and it was working as intended
